pubsub/aws/snssqs/snssqs.go

```diff
@@ -343,20 +416,36 @@ func (s *snsSqs) getOrCreateQueue(queueName string) (*sqsQueueInfo, error) {
 	return queueInfo, nil
 }
 
+func (s *snsSqs) getMessageGroupID(req *pubsub.PublishRequest) *string {
+	if len(s.metadata.fifoMessageGroupID) > 0 {
+		return &s.metadata.fifoMessageGroupID
+	}
+	// each daprd, of a given PubSub, of a given publisher application publishes to a message group ID of its own.
+	// for example: for a daprd serving the SNS/SQS Pubsub component we generate a unique id -> A; that component serves on behalf
+	// of a given PubSub deployment name B, and component A publishes to SNS on behalf of a dapr application named C (effectively to topic C).
+	// therefore the created message group ID for publishing messages in the aforementioned setup is "A:B:C".
```

Comment:
Again, also probably useful information to include in https://github.com/dapr/docs/issues/1977, it's generally helpful for dev debugging to understand how inputs map to underlying representations that can show up in tracing or the service backend logs.